src/eos-metrics-instrumentation.c

```diff
@@ -111,6 +112,21 @@
 #define LIVE_BOOT_FLAG_REGEX "\\bendless\\.live_boot\\b"
 #define DUAL_BOOT_FLAG_REGEX "\\bendless\\.image\\.device\\b"
 
+/*
+ * Recorded once at startup to report the image ID. This is a string such as
+ * "eos-eos3.1-amd64-amd64.170115-071322.base" which is saved in an attribute
+ * on the root filesystem by the image builder, and allows us to tell the
+ * channel that the OS was installed by (eg download, OEM pre-install, Endless
+ * hardware, USB stick, etc) and which version was installed. The payload
+ * is a single string containing this image ID, if present.
```

Comment:
Seems implicit by the fact that all metrics events must be valid GVariants, so any "string" is defined to be UTF-8.